### PR TITLE
Stringify response before json.loads

### PR DESCRIPTION
**Pull request by AroundTheLines**
This fixes the "the json object must be str, not 'bytes'" error that happens on line 145.

Sample error before fix:

```
Traceback (most recent call last):
  File "(...)/twitter.py", line 53, in <module>
    async_data = LineItem.async_stats_job_data(account, async_stats_job_result['url'])
  File "(...)/venv/lib/python3.5/site-packages/twitter_ads/resource.py", line 261, in async_stats_job_data
    raw_body=True, stream=True).perform()
  File "(...)/venv/lib/python3.5/site-packages/twitter_ads/http.py", line 73, in perform
    response = self.__oauth_request()
  File "(...)/venv/lib/python3.5/site-packages/twitter_ads/http.py", line 103, in __oauth_request
    body=response.raw, raw_body=raw_response_body)
  File "(...)/venv/lib/python3.5/site-packages/twitter_ads/http.py", line 145, in __init__
    self._body = json.loads(raw_response_body)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
```

### CODE REVIEW

File: twitter_ads/http.py [hunk 1]
```diff
@@ -141,6 +141,7 @@ def __init__(self, code, headers, **kwargs):
             raw_response_body = self._raw_body
 
         try:
+            raw_response_body = str(raw_response_body)
```
Comment (frewsxcv) [inline twitter_ads/http.py]: This probably doesn't need to be in the `try`, unless `str(..)` raises a `ValueError`, which I don't think it does? Not that big of a deal though.

Comment (AroundTheLines) [inline twitter_ads/http.py]: That was how I found the code, so I didn't change that. I did just change location of the stringify just now though.

File: twitter_ads/http.py [hunk 2]
```diff
@@ -141,6 +141,8 @@ def __init__(self, code, headers, **kwargs):
             raw_response_body = self._raw_body
 
         try:
+            if not isinstance(raw_response_body, str):
```
Comment (frewsxcv) [inline twitter_ads/http.py]: Under what scenario would `raw_response_body` not be a `str`?

Comment (frewsxcv) [inline twitter_ads/http.py]: Just thinking out loud, if it's a `str` only when `zlib.decompress` gets called (a few lines above this), we should maybe move the `decode()` up in that conditional.

Comment (AroundTheLines) [inline twitter_ads/http.py]: I'm not entirely sure if that's the case, I'll test it out and get back to you on that.

**Comment by itsamaik** [comment]
Please commit this change into the lastest version, currently I'm also getting this error when retrieving job results. After zlib.decompress it's returning b'{ .......... }' thus resulting into an error when it's put into json.loads().

**Comment by jbabichjapan** [comment]
@AroundTheLines Hey could you check the latest feedback and let us know if you are ready to go with this change? I don't have time to help drive this one in right now but if you need more eyeballs please call out and hopefully some people will notice and help review

**Comment by juanshishido** [comment]
As @frewsxcv suggested, I added `.decode()` to [line 139](https://github.com/twitterdev/twitter-python-ads-sdk/blob/master/twitter_ads/http.py#L139):

```python
raw_response_body = zlib.decompress(self._raw_body, 16 + zlib.MAX_WBITS).decode()
```

Async stats working locally for both Python 2.7.12 and Python 3.5.2.

Any thoughts on this, @AroundTheLines?

cc: @jbabichjapan, @tushdante

**Comment by juanshishido** [comment]
@AroundTheLines: We'd like to merge this fix. Please let us know if you're able to make the aforementioned changes. If not, we can do it, too. Thanks for bringing this up and proposing a solution!

**Comment by AroundTheLines** [comment]
@juanshishido: Hey, sorry I haven't been keeping up with this PR. I'd be glad to make the changes in a moment.

**Comment by juanshishido** [comment]
Awesome! Thanks, @AroundTheLines! 👍

**Comment by madman-bob** [comment]
I'm having this issue sporadically in version 1.2.2.

Eyeballing the source, I'm guessing we also need a `.decode()` in `http.py:97`. ie. Changing
`raw_response_body = response.raw.read() if stream else response.text`
to
`raw_response_body = response.raw.read().decode() if stream else response.text`.

Can anyone more knowledgable than I confirm/deny?

**Pull request merged**